### PR TITLE
MSQ Reindex IT

### PR DESCRIPTION
**Pull request by abhagraw**
Migrate current integration indexer tests to equivalent MSQ tests in the new IT Framework.

### CODE REVIEW

File: integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/AbstractITSQLBasedIngestion.java [hunk 1]
```diff
@@ -46,6 +52,28 @@
   @Inject
   private DataLoaderHelper dataLoaderHelper;
 
+  @Rule
+  public TestWatcher watchman = new TestWatcher()
+  {
+    @Override
+    public void starting(Description d)
+    {
+      System.out.printf("\n\nRunning %s\n", d.getDisplayName());
```
Comment (cryptoe) [inline integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/AbstractITSQLBasedIngestion.java:61]: Lets use Log here.

File: integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/ITMSQReindexTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        Arrays.asList("wikipedia_index_msq.sql", "wikipedia_reindex_msq.sql", "wikipedia_reindex_queries.json"),
+        Arrays.asList("wikipedia_merge_index_msq.sql", "wikipedia_merge_reindex_msq.sql", "wikipedia_merge_index_queries.json"),
+        Arrays.asList("wikipedia_index_task_with_transform.sql", "wikipedia_reindex_with_transform_msq.sql", "wikipedia_reindex_queries_with_transforms.json")
+    );
+
+  }
+
+  @Test
+  @Parameters(method = "test_cases")
+  @TestCaseName("Test_{index} ({0}, {1}, {2})")
```
Comment (cryptoe) [inline integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/ITMSQReindexTest.java:59]: 🚀

File: integration-tests/src/main/java/org/apache/druid/testing/utils/DataLoaderHelper.java [hunk 3]
```diff
@@ -33,17 +35,21 @@
 
   public void waitUntilDatasourceIsReady(String datasource)
   {
+    LOG.info("Waiting for Segments to load");
```
Comment (cryptoe) [inline integration-tests/src/main/java/org/apache/druid/testing/utils/DataLoaderHelper.java:38]: add datasource names here too

File: integration-tests/src/main/java/org/apache/druid/testing/utils/SqlTestQueryHelper.java [hunk 4]
```diff
@@ -59,9 +59,11 @@ public boolean isDatasourceLoadedInSQL(String datasource)
         null
     );
 
+    LOG.info("Running sample query.");
     try {
       //noinspection unchecked
       queryClient.query(getQueryURL(broker), query);
+      LOG.info("Sample query successful.");
```
Comment (cryptoe) [inline integration-tests/src/main/java/org/apache/druid/testing/utils/SqlTestQueryHelper.java:66]: Nit: I think we do not need logging in this class.

File: integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/AbstractITSQLBasedIngestion.java [hunk 5]
```diff
@@ -46,6 +52,28 @@
...
+    @Override
+    public void failed(Throwable e, Description d)
+    {
+      LOG.error("%s FAILED\n", d.getDisplayName());
+    }
+
+    @Override
+    public void finished(Description d)
+    {
+      LOG.info("%s finished\n", d.getDisplayName());
```
Comment (cryptoe) [inline integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/AbstractITSQLBasedIngestion.java:73]:
```suggestion
      LOG.info("FINISHED %s", d.getDisplayName());
```
minor nit.

File: integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/AbstractITSQLBasedIngestion.java [hunk 6]
```diff
@@ -46,6 +52,28 @@
...
+    @Override
+    public void starting(Description d)
+    {
+      LOG.info("\n\n\nRunning %s", d.getDisplayName());
+    }
+
+    @Override
+    public void failed(Throwable e, Description d)
+    {
+      LOG.error("%s FAILED\n", d.getDisplayName());
```
Comment (cryptoe) [inline integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/AbstractITSQLBasedIngestion.java:67]:
```suggestion
      LOG.error("FAILED %s", d.getDisplayName());
```

File: integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/AbstractITSQLBasedIngestion.java [hunk 7]
```diff
@@ -46,6 +52,28 @@
   @Inject
   private DataLoaderHelper dataLoaderHelper;
 
+  @Rule
+  public TestWatcher watchman = new TestWatcher()
+  {
+    @Override
+    public void starting(Description d)
+    {
+      LOG.info("\n\n\nRunning %s", d.getDisplayName());
```
Comment (cryptoe) [inline integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/AbstractITSQLBasedIngestion.java:61]:
```suggestion
      LOG.info("RUNNING %s", d.getDisplayName());
```

File: integration-tests/src/main/java/org/apache/druid/testing/utils/SqlTestQueryHelper.java [hunk 8]
```diff
@@ -59,6 +59,7 @@ public boolean isDatasourceLoadedInSQL(String datasource)
         null
     );
 
+    LOG.info("Verifying datasource [%s] availability for SQL queries", datasource);
```
Comment (cryptoe) [inline integration-tests/src/main/java/org/apache/druid/testing/utils/SqlTestQueryHelper.java:62]: I think this log line is not needed as we already have one in DataLoadHelper#48

**Review by cryptoe — APPROVED** [review]
Left minor comments. LGTM otherwise. 
Will merge after CICD is green.

**Comment by abhagraw** [comment]
@cryptoe Can this be merge? Will pick up the nits in my next PR.

**Comment by abhagraw** [comment]
Build running again, because I closed the PR by mistake. It is already green.

**Comment by cryptoe** [comment]
Thanks, @abhagraw for the contribution.

**Pull request merged**